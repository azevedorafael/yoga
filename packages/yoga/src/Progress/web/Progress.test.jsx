import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, Progress } from '../..';

describe('<Progress />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Progress />
        <Progress max="100" />
        <Progress max="100" value="60" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with label string', () => {
    const { container } = render(
      <ThemeProvider>
        <Progress max="100" label={{ value: 'Some decription here' }} />
        <Progress
          max="100"
          value="65"
          label={{ value: 'Some decription here', placement: 'right' }}
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with label number', () => {
    const { container } = render(
      <ThemeProvider>
        <Progress max="100" value="20" label={{ value: 20 }} />
        <Progress
          max="100"
          value="65"
          label={{ value: 65, placement: 'right' }}
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
