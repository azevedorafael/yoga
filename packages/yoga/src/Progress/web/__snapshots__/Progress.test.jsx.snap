// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> should match snapshot 1`] = `
.c1 {
  border: none;
  width: 100%;
  height: 8px;
}

.c1.c1,
.c1::-webkit-progress-value,
.c1[value]::-webkit-progress-bar,
.c1:not(value)::-webkit-progress-bar {
  border-radius: 8px;
}

.c1:not(value)::-moz-progress-bar,
.c1[value]::-moz-progress-bar {
  border-radius: 8px;
}

.c1:not(value)::-webkit-progress-bar {
  background-color: #F5F5FA;
}

.c1::-webkit-progress-value {
  background-color: #F46152;
}

.c1.c1,
.c1:not(value)::-moz-progress-bar {
  background-color: #F5F5FA;
}

.c1[value]::-moz-progress-bar {
  background-color: #F46152;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .c2 {
  margin-top: 4px;
  max-width: 280px;
}

<div>
  <div
    class="c0"
  >
    <progress
      class="c1"
      max="1"
    />
  </div>
  <div
    class="c0"
  >
    <progress
      class="c1"
      max="100"
    />
  </div>
  <div
    class="c0"
  >
    <progress
      class="c1"
      max="100"
      value="60"
    />
  </div>
</div>
`;

exports[`<Progress /> should match snapshot with label number 1`] = `
.c2 {
  border: none;
  width: 100%;
  height: 8px;
}

.c2.c2,
.c2::-webkit-progress-value,
.c2[value]::-webkit-progress-bar,
.c2:not(value)::-webkit-progress-bar {
  border-radius: 8px;
}

.c2:not(value)::-moz-progress-bar,
.c2[value]::-moz-progress-bar {
  border-radius: 8px;
}

.c2:not(value)::-webkit-progress-bar {
  background-color: #F5F5FA;
}

.c2::-webkit-progress-value {
  background-color: #F46152;
}

.c2.c2,
.c2:not(value)::-moz-progress-bar {
  background-color: #F5F5FA;
}

.c2[value]::-moz-progress-bar {
  background-color: #F46152;
}

.c4 {
  font-size: 12px;
}

.c6 .c3 {
  margin-top: 4px;
  max-width: 280px;
}

.c7 .c3 {
  margin-top: 4px;
  max-width: 280px;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 .c3 {
  width: 22px;
  margin-right: 8px;
}

.c0 .c1 {
  margin: auto;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  text-align: right;
}

.c5 .c3 {
  width: 22px;
  margin-left: 8px;
}

.c5 .c1 {
  margin: auto;
}

<div>
  <div
    class="c0"
  >
    <progress
      class="c1 c2"
      max="100"
      value="20"
    />
    <label
      class="c3 c4"
    >
      20
    </label>
  </div>
  <div
    class="c5"
  >
    <progress
      class="c1 c2"
      max="100"
      value="65"
    />
    <label
      class="c3 c4"
    >
      65
    </label>
  </div>
</div>
`;

exports[`<Progress /> should match snapshot with label string 1`] = `
.c1 {
  border: none;
  width: 100%;
  height: 8px;
}

.c1.c1,
.c1::-webkit-progress-value,
.c1[value]::-webkit-progress-bar,
.c1:not(value)::-webkit-progress-bar {
  border-radius: 8px;
}

.c1:not(value)::-moz-progress-bar,
.c1[value]::-moz-progress-bar {
  border-radius: 8px;
}

.c1:not(value)::-webkit-progress-bar {
  background-color: #F5F5FA;
}

.c1::-webkit-progress-value {
  background-color: #F46152;
}

.c1.c1,
.c1:not(value)::-moz-progress-bar {
  background-color: #F5F5FA;
}

.c1[value]::-moz-progress-bar {
  background-color: #F46152;
}

.c3 {
  font-size: 12px;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .c2 {
  margin-top: 4px;
  max-width: 280px;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  text-align: right;
}

.c4 .c2 {
  margin-top: 4px;
  max-width: 280px;
}

<div>
  <div
    class="c0"
  >
    <progress
      class="c1"
      max="100"
    />
    <label
      class="c2 c3"
    >
      Some decription here
    </label>
  </div>
  <div
    class="c4"
  >
    <progress
      class="c1"
      max="100"
      value="65"
    />
    <label
      class="c2 c3"
    >
      Some decription here
    </label>
  </div>
</div>
`;
